### DISCUSSION TITLE
Configure interpreter based on python environment

### DISCUSSION THREAD

**Discussion by albas99** [zed-industries/zed#6511] [Feature Requests]
I'm using Zed along with the Warp terminal app, and even though I've configured my virtual environment using Poetry, Zed is still flagging my code and doesn't seem to be recognizing the interpreter. How can I configure it so it can recognize it?

UPDATE: I've managed to configure pyright in my `pyproject.toml` file but it's still not working. I'm still getting `import django.contrib not resolved` and I have django installed in the poetry environment.
![Screenshot 2023-03-10 at 12 41 19](https://user-images.githubusercontent.com/53338885/224253722-0e73e970-ed45-4ad8-bcd6-9b663a2b1a16.png)

**Comment by JosephTLyons** [comment]
We use pyright under the hood - I'm not quite sure what one might need to do to get pyright and poetry to communicate, but one thing you could do is make a `pyrightconfig.json` file in your repo:

```json
{
  "venvPath": ".",
  "venv": ".venv"
}
```

Where `venvPath` is the venv path relative to this file
Where `venv ` is name of venv directory

https://docs.zed.dev/languages/python

At some point, we should work on automatically detecting virtual environments.

**Reply by albas99** [reply]
I did that but in my `pyproject.toml` configuration, the issue probably has something to do with the location of my poetry environment(in `Library/Caches/`), I'm not sure though, but at the moment it seems to have recognized the config(also not sure what I did that fixed it😅).

**Reply by natemcintosh** [reply]
I am having the same issue. Poetry's default venv placement is in `~/Library/Caches/..` and even though I've specified the relative path as the documentation suggests (I've also tried the absolute path) it still is not connecting to the venv

**Update**
I was able to get it to work eventually. The issue was that I did not read the documentation closely enough. I thought the `venvPath` was the path to the venv, but it is actually the path to the parent folder of the venv. Then the `venv` argument is the name of the venv in the parent folder. I would have understood this if I'd read more closely, but I also found it rather surprising. I would have expected that `venvPath` was the full path to the venv (not to the parent) and that `venv` was just what I wanted it to be displayed as.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
